File: typescript/4.ts
let currentMax = 0;
for (let i = 999; i >= 100; i--) {
    for (let j = i; j >= 100; j--) {
        const numStr = (i * j).toString();
        if (numStr === numStr.split("").reverse().join("") && i * j > currentMax) {
            currentMax = i * j;
        }
    }
}

console.log(currentMax);
// 906609

export {}

File: swift/4.swift
var currentMax = 0
for i in stride(from: 999, to: 100, by: -1) {
  for j in stride(from: i, to: 100, by: -1) {
    let numStr = "\(i*j)"
    if String(numStr.reversed()) == numStr && i * j > currentMax {
      currentMax = i*j
    }
  }
}

print(currentMax)
// 906609
